linux/userdata_common.sh

```diff
@@ -26,6 +24,34 @@ debug-2s3() {
   aws s3 cp "${tfi_userdata_log}" "s3://$build_slug/$${index_str}$${ami_key}/" || true
 }
 
+check-metadata-availability() {
+  ## This will not return until metadata is available.
+  local metadata_loopback_az="http://169.254.169.254/latest/meta-data/placement/availability-zone"
+
+  while :
```

Comment:
I'm not sure this syntax is correct. When I try it locally, I just get a syntax error. But I think the whole thing can be simplified anyway...

```
check-metadata-availability() {
  ## This will not return until metadata is available.
  local metadata_loopback_az="http://169.254.169.254/latest/meta-data/placement/availability-zone"

  while ! curl -sSL $metadata_loopback_az
  do
    debug-2s3 "$(date +%F_%T): ERROR connecting to ec2 metadata. Will try again."
    sleep 1
  done

  debug-2s3 "$(date +%F_%T): Connected to ec2 metadata successfully"
}
```

or use the `retry` function already available here...

```
check-metadata-availability() {
  local metadata_loopback_az="http://169.254.169.254/latest/meta-data/placement/availability-zone"

  retry 50 curl -sSL $metadata_loopback_az

  debug-2s3 "$(date +%F_%T): Connected to ec2 metadata successfully"
}
```

Reply:
Syntax works fine but I'm always happy to simplify. So far I haven't had any issues with the Linux boxes. Windows 19 is not getting any metadata ever.